diff --git a/dropbear_install b/dropbear_install
index 40a5116..3180883 100644
--- a/dropbear_install
+++ b/dropbear_install
@@ -8,7 +8,7 @@ get_fingerprint() {
 display_fingerprints() {
   local keyfile
 
-  for keyfile in "/etc/dropbear/dropbear_rsa_host_key" "/etc/dropbear/dropbear_dss_host_key" "/etc/dropbear/dropbear_ecdsa_host_key" ; do
+  for keyfile in "/etc/dropbear/dropbear_rsa_host_key" "/etc/dropbear/dropbear_ecdsa_host_key" ; do
       if [ -s "${keyfile}" ] ; then
 	  echo "$(basename "${keyfile}") : $(get_fingerprint "${keyfile}")"
       fi
@@ -17,7 +17,6 @@ display_fingerprints() {
 
 copy_openssh_keys() {
   local osshrsa="/etc/ssh/ssh_host_rsa_key"
-  local osshdsa="/etc/ssh/ssh_host_dsa_key"
   local osshecdsa="/etc/ssh/ssh_host_ecdsa_key"
 
   local dbpre="/etc/dropbear/dropbear_"
@@ -29,11 +28,6 @@ copy_openssh_keys() {
       return_code=0
   fi
 
-  if [ -s "$osshdsa" ]; then
-      dropbearconvert openssh dropbear $osshdsa ${dbpre}dss_host_key
-      return_code=0
-  fi
-
   if [ -s "$osshecdsa" ]; then
       dropbearconvert openssh dropbear $osshecdsa ${dbpre}ecdsa_host_key
       return_code=0
@@ -44,7 +38,7 @@ copy_openssh_keys() {
 
 generate_keys() {
   local keyfile keytype
-  for keytype in rsa dss ecdsa ; do
+  for keytype in rsa ecdsa ; do
       keyfile="/etc/dropbear/dropbear_${keytype}_host_key"
       if [ ! -s "$keyfile" ]; then
 	  echo "Generating ${keytype} host key for dropbear ..."
